src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetection.swift

```diff
@@ -27,7 +27,7 @@ final class ExposureDetection {
 	private var completion: Completion?
 	private var progress: Progress?
 	private var countryKeypackageDownloader: CountryKeypackageDownloading
-	private let appConfigurationProvider: AppConfigurationProviding
+	private let appConfiguration: SAP_ApplicationConfiguration
```

Comment:
Why are we losing the protocol here?

Reply:
Because we can directly inject the actual App Configuration instead of the provider that tries to get it from the cache. The only caller of the exposure detection's initializer retrieves a cached app configuration beforehand anyway. That way we can ensure that all involved instances work with the same configuration. And we were able to make the exposure detection's dependency on the app configuration more clear by passing it in.